compiler-rt/lib/builtins/aarch64/sme-libc-routines.c

```diff
@@ -1,5 +1,20 @@
 #include <stddef.h>
 
+#if __ARM_FP == 0
+// WARNING: When building the scalar versions of these functions you need to
+// use the compiler flag "-mllvm -disable-loop-idiom-all" to prevent clang
```

Comment:
I think `-fno-builtin-memcpy` would be a better option to recommend here, as that's a user-facing option, while the  `-mllvm` options are internal options, mostly intended for LLVM developers.

Could this be added to the CMakeLists.txt for this file?

Reply:
I just checked -- the CMakeLists.txt file already uses `-fno-builtins`:
```    
set_source_files_properties(aarch64/sme-libc-routines.c PROPERTIES COMPILE_FLAGS "-fno-builtin")
```
I've removed this mis-leading comment; we need `-fno-builtin` for every function in this file, and the memchr version doesn't talk about it.